tree/homework2.php

```diff
@@ -0,0 +1,84 @@
+<?php
+require_once ('homework1.php');
+
+class Queue
+{
+    private $elements;
+
+    public function __construct()
+    {
+        $this->elements = array();
+    }
+
+    public function enqueue($num) {
+        array_unshift($this->elements, $num);
+    }
+
+    public function dequeue() {
+        if (!$this->isEmpty()) {
+            unset($this->elements[sizeof($this->elements) - 1]);
```

Comment:
It should return the last element and remove it from the queue
```php
if (!$this->isEmpty()) {
    return array_pop($this->elements);
}
```

Reply:
Thanks for pointing out that.